File: server/EnglishTutorAI.Application/Models/CompletedTranslateListItem.cs
using EnglishTutorAI.Domain.Enums;

namespace EnglishTutorAI.Application.Models;

public class CompletedTranslateListItem
{
    public Guid TranslateId { get; set; }

    public required string Title { get; set; }

    public StudyTopic StudyTopic { get; set; }

    public required string Content { get; set; }

    public DateTime? CompletedOn { get; set; }
}

File: client/src/app/dataModels/translate/completedTranslateListItem.ts
import {StudyTopic} from "@/app/dataModels/enums/studyTopic.ts";

export interface CompletedTranslateListItem {
    translateId: string;
    title: string;
    studyTopic: keyof typeof StudyTopic;
    content: string;
    completedOn: string;
}